### PR TITLE
[#292] Update TF version and GH actions versions

### LINKED ISSUE #292: Update TF version and GH actions versions
## Why

It is a good practice to keep dependency updated.

## Who Benefits?

Developers

### PR DESCRIPTION
**Pull request by Nihisil**
- Close #292

## What happened 👀

- Upgrade TF to 1.8.3 version
- Upgrade GH actions to latest versions
- Install TF and node through asdf instead of manual installation

## Proof Of Work 📹

CI is green.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
         run: npm run test '--ignore-scripts' -- --coverage --ci
 
       - name: Upload test coverage
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```
Comment (longnd) [inline .github/workflows/test.yml:40]: Please limit the [retention period](https://github.com/actions/upload-artifact?tab=readme-ov-file#retention-period) for the artifact (default is [90 days](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization)). Storing the artifact is counted toward the storage limit which [affect the cost](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes).

**Review by longnd — APPROVED** [review]
It is a cleaner way to install dependencies. I will borrow it to other projects I'm working on 🤩

File: .github/workflows/test.yml (hunk 1 above)
Comment (Nihisil) [reply .github/workflows/test.yml:40]: Good catch, we differently don't want to have artifacts to be stored for 90 days. I have set it to 3 days here: https://github.com/nimblehq/infrastructure-templates/pull/293/commits/ece63c64444ed0e7d01c7b06d6bcb81a638d5c16

**Comment by Nihisil** [comment]
> It is a cleaner way to install dependencies. I will borrow it to other projects I'm working on 🤩

Yep, I like it as well. It was introduced by @nvminhtue in his PR: https://github.com/nimblehq/infrastructure-templates/pull/262

**Review by malparty — APPROVED** [review]
Nice 🚀

**Review by bterone — APPROVED** [review]

**Review by hoangmirs — APPROVED** [review]

**Review by sanG-github — APPROVED** [review]

**Review by nvminhtue — APPROVED** [review]

**Pull request merged**